apps/common/lib/lexical/code_unit.ex

```diff
@@ -1,63 +1,36 @@
 defmodule Lexical.CodeUnit do
   @moduledoc """
-  Code unit and offset conversions
-
-  The LSP protocol speaks in positions, which defines where something happens in a document.
-  Positions have a start and an end, which are defined as code unit _offsets_ from the beginning
-  of a line. this module helps to convert between utf8, which most of the world speaks
-  natively, and utf16, which has been forced upon us by microsoft.
-
-  Converting between offsets and code units is 0(n), and allocations only happen if a
-  multi-byte character is detected, at which point, only that character is allocated.
-  This exploits the fact that most source code consists of ascii characters, with at best,
-  sporadic multi-byte characters in it. Thus, the vast majority of documents will not require
-  any allocations at all.
+  Code unit and offset conversions.
+
+  LSP positions are encoded as UTF-16 code unit offsets from the beginning of a line,
+  while positions in Elixir are UTF-8 character positions (graphemes). This module
+  deals with converting between the two.
   """
-  @type utf8_code_unit :: non_neg_integer()
-  @type utf16_code_unit :: non_neg_integer()
-  @type utf8_offset :: non_neg_integer()
-  @type utf16_offset :: non_neg_integer()
 
-  @type error :: {:error, :misaligned} | {:error, :out_of_bounds}
+  @type utf8_character_position :: non_neg_integer()
+  @type utf8_code_unit_offset :: non_neg_integer()
+  @type utf16_code_unit_offset :: non_neg_integer()
 
-  # public
+  @type error :: {:error, :misaligned} | {:error, :out_of_bounds}
 
   @doc """
-  Converts a utf8 character offset into a utf16 character offset. This implementation
-  clamps the maximum size of an offset so that any initial character position can be
-  passed in and the offset returned will reflect the end of the line.
+  Converts a 0-based UTF-8 character position to a UTF-16 code unit offset.
   """
-  @spec utf16_offset(String.t(), utf8_offset()) :: utf16_offset()
-  def utf16_offset(binary, character_position) do
+  @spec utf8_char_to_utf16_offset(String.t(), utf8_character_position()) ::
+          utf16_code_unit_offset()
+  def utf8_char_to_utf16_offset(binary, character_position) do
```

Comment:
not a fan of the new name, `utf8` doesn't have the notion of `char`. Suggestion is `utf8_column_to_utf16_offset` or `utf8_cursor_to_utf16_offset`

Reply:
How about `utf8_position_to_utf16_offset`?